File: RayTrace_OfflineEditor/OfflineEditor/OfflineEditor/Sphere.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OfflineEditor
{
    class Sphere
    {
        public Vec3f Center;
        public Vec3f Velocity;
        public float Radius;
        public Vec3f SurfaceColor, EmissionColor;
        public float Transparency, Reflection;
        public float RadiusIncrement;

        public Sphere(  Vec3f surfaceColor = default , Vec3f center = default, Vec3f velocity = default, Vec3f emissionColor = default,
            float radius = 0.0f, float transparency = 0.0f, float reflection = 0.0f, float radiusIncrement = 0.0f) 
        {
            Center = center;
            Velocity = velocity;
            Radius = radius;
            SurfaceColor = surfaceColor;
            EmissionColor = emissionColor;
            Transparency = transparency;
            Reflection = reflection;
            RadiusIncrement = radiusIncrement;
        }
    }
}

File: RayTrace_OfflineEditor/RayTracerFramework/RayTracerSmall/Sphere.cpp
#include "Sphere.h"
#include <iostream>

Heap* Sphere::s_pHeap = nullptr;

Sphere::Sphere(const Vec3f& c,
	const float& r,
	const Vec3f& sc,
	const float& refl ,
	const float& transp ,
	const Vec3f & ec) : center(c), radius(r), radius2(r* r), surfaceColor(sc), emissionColor(ec),
	transparency(transp), reflection(refl)
{
	//std::cout << "Sphere created" << std::endl;#
}

Sphere::~Sphere()
{
	//std::cout << "Sphere deleted" << std::endl;
}

void Sphere::SetRadius2(float increment)
{
	radiusIncrement = increment;
}

void Sphere::Update(int frame)
{

}

void* Sphere::operator new(size_t size)
{
	std::cout << "new being called" << std::endl;
	if (s_pHeap == nullptr)
	{
		s_pHeap = HeapFactory::CreateHeap("Sphere Object");
	}
	return ::operator new(size, s_pHeap);
}

void Sphere::operator delete(void* p, size_t size)
{
	::operator delete(p);
}